task/local_db/refactor_processor

teamagam5 opened the pull request:
moved logics to be after repositories

teamagam5 committed:
Refactored the VectorLayer process.
Now the cache and the poller add VLs to the repository and the ProcessVectorLayerInteractor is a background task launched by the LauncherActivity and attached to the VLRepo's observable.

teamagam5 committed:
Replaced the AddNewMessageToRepositoryInteractor with UpdateUnreadCountInteractor which is now also in the background.
Classes that used the old interactor now just put messages in the MessagesRepository.
The Processor yet has its visitor that should be moved to an external background interactor.

teamagam5 committed:
Finished refactor, now the processor only puts the items to their correlative repository.
Introduced one final background-running interactor: ProcessMessageInteractor.

teamagam5 committed:
Removed an unnecessary interactor,
Added Test class for the new interactor.

BarAgam commented on domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractor.java:
```diff
@@ -0,0 +1,127 @@
+                .retryWhen(new RetryWithDelay(Constants.LAYER_CACHING_RETRIES,
+                    Constants.LAYER_CACHING_RETRIES_DELAY_MS))
+                .doOnError(this::logFailure)
+                .onErrorResumeNext(Observable.empty()));
```
We need to see that now this line doesn't prevent the processing of future vector layers

teamagam5 replied:
If you'd have known how much trouble you caused by this little cute observation... Blimey!

teamagam5 replied:
PLS re-CR the last commit

BarAgam reviewed (approved):
Bravo

teamagam5 committed:
CR Fixes.
Added a Test class for ProcessVectorLayerInteractor.
The test tests only 2 aspects of the interactor, concerning the caching mechanism.

BarAgam commented on domain/src/test/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractorCachingTest.java:
```diff
@@ -0,0 +1,68 @@
+  }
+
+  @Test
+  public void whenCannotCacheVectorLayer_InteractorIgnoresItAndContinues() throws Exception {
```
<3
